gpytorch/kernels/grid_kernel.py

```diff
@@ -61,16 +61,14 @@ def __init__(
         if not self.interpolation_mode:
             self.register_buffer("full_grid", create_data_from_grid(grid))
 
+    def _clear_cache(self):
+        del self._cached_kernel_mat
+
     def register_buffer_list(self, base_name, tensors):
         """Helper to register several buffers at once under a single base name"""
         for i, tensor in enumerate(tensors):
             self.register_buffer(base_name + "_" + str(i), tensor)
 
-    def train(self, mode=True):
-        if hasattr(self, "_cached_kernel_mat"):
```

Comment:
Were these `hasattr` checks unnecessary?

Reply:
Oh good call - those checks should be added back in.

Reply:
Fixed.